tests/test_generic_camera.py

```diff
@@ -316,3 +317,125 @@ async def test_take_cwfs(self) -> None:
 
     async def test_take_acq(self) -> None:
         await self.assert_take_acq()
+
+    image_types: Dict[str, Dict[str, Any]] = {
+        "BIAS": {
+            "assert_method": "assert_take_bias",
+            "params": {"nbias": 1},
+        },
+        "DARK": {
+            "assert_method": "assert_take_darks",
+            "params": {"ndarks": 1, "exptime": 1.0},
+        },
+        "FLAT": {
+            "assert_method": "assert_take_flats",
+            "params": {"nflats": 1, "exptime": 1.0},
+        },
+        "FOCUS": {
+            "assert_method": "assert_take_focus",
+            "params": {"n": 1, "exptime": 1.0},
+        },
+        "OBJECT": {
+            "assert_method": "assert_take_object",
+            "params": {"n": 1, "exptime": 1.0},
+        },
+        "ENGTEST": {
+            "assert_method": "assert_take_engtest",
+            "params": {"n": 1, "exptime": 1.0},
+        },
+        "ACQ": {
+            "assert_method": "assert_take_acq",
+            "params": {"n": 1, "exptime": 1.0},
+        },
+        "CWFS": {
+            "assert_method": "assert_take_cwfs",
+            "params": {"n": 1, "exptime": 1.0},
+        },
+    }
+
+    def reset_mocks(self) -> None:
+        self.remote_group.camera.cmd_takeImages.set.reset_mock()
+        self.remote_group.camera.cmd_takeImages.start.reset_mock()
+        self.remote_group.camera.evt_endReadout.flush.reset_mock()
+
+    async def test_take_image_no_tcs_sync(self) -> None:
+        image_types_no_tcs = ["BIAS", "DARK", "FLAT", "FOCUS"]
+
+        for img_type in image_types_no_tcs:
+            with self.subTest(image_type=img_type):
+                with self.get_fake_tcs() as fake_tcs:
+                    fake_tcs.fail = False
+                    methods = self.image_types.get(img_type)
+                    if not methods:
+                        self.fail(
```

Comment:
Why not just raise an exception? You could do something like

```
assert not methods, f"Image type '{img_type}' not found in image_types mapping."
```

Reply:
ok, let me try that, I was trying to get around a mypy issue, that was thrown on the method signature type.